docs/docs/recipes/integrate-logto/spring-boot.mdx

```diff
@@ -0,0 +1,226 @@
+import org.springframework.security.oauth2.client.web.DefaultOAuth2AuthorizationRequestResolver;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestCustomizers;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestResolver;
+import org.springframework.security.web.SecurityFilterChain;
+
+@EnableWebSecurity
+public class SecurityConfiguration {
+
+	@Bean
+	public OAuth2AuthorizationRequestResolver oauth2AuthorizationRequestResolverWithPkce(
+			ClientRegistrationRepository clientRegistrationRepository
+	) {
+		DefaultOAuth2AuthorizationRequestResolver authorizationRequestResolver =
+				new DefaultOAuth2AuthorizationRequestResolver(clientRegistrationRepository, "/oauth2/authorization");
+		authorizationRequestResolver.setAuthorizationRequestCustomizer(OAuth2AuthorizationRequestCustomizers.withPkce());
+		return  authorizationRequestResolver;
+	}
+
+	@Bean
+	public SecurityFilterChain securityFilterChain(
+			HttpSecurity http,
+			OAuth2AuthorizationRequestResolver oauth2AuthorizationRequestResolverWithPkce
+	) throws Exception {
+		http.authorizeRequests()
+		    // Anyone can access the home page.
+		    // (At least one mapping is required. Change it as you need.)
+        .mvcMatchers("/").permitAll()
+				.and().oauth2Login()
+				.authorizationEndpoint()
+				.authorizationRequestResolver(oauth2AuthorizationRequestResolverWithPkce);
+		return http.build();
+	}
+}
+```
+
+</details>
+
+### Implement sign-in in route
+
+A user can visit the route `/oauth2/authorization/logto` (e.g. `http://localhost:3000/oauth2/authorization/logto`) to sign in.
+Spring Security will use the `spring.security.oauth2.client` configuration defined above to handle user sign-in automatically.
+
+Let's try this sign-in link on your web application page.
+
+Add a home page template file:
+
+```html
+<!-- path/to/project/src/main/resources/templates/index.html -->
+<!DOCTYPE html>
+<html
+  lang="en"
+  xmlns:th="http://www.thymeleaf.org"
+  xmlns:sec="http://www.thymeleaf.org/thymeleaf-extras-springsecurity5"
+>
+  <body>
+    <div sec:authorize="!isAuthenticated()">
+      <a th:href="@{/oauth2/authorization/logto}">Sign in</a>
+    </div>
+    <div sec:authorize="isAuthenticated()">
+      <p>Signed in</p>
+    </div>
+  </body>
+</html>
+```
+
+Add a controller to render the view:
+
+```java
+package io.logto.springboot.sample.controller;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class HomeController {
+
+    @GetMapping("/")
+    public String home() {
+        return "index";
+    }
+}
+```
+
+Run your Spring Boot web application, visit the home page [http://localhost:3000/](http://localhost:3000/), click the _Sign in_ link on it, and it will redirect to the Logto sign-in page.
+
+After the user signs in successfully, it will redirect back to the home page showing "Signed in" that means the user is authenticated.
```

Comment:
```suggestion
Provide your credentials, and once authenticated, it will redirect you back to the home page with a message that shows "Signed in".
```